code/game/gamemodes/nuclear/pinpointer.dm

```diff
@@ -361,6 +361,13 @@
 	if(istype(I, /obj/item/gun/energy/detective))
 		link_gun(I.UID())
 
+/obj/item/pinpointer/crew/emp_act(severity)
+	var/obj/item/gun/energy/detective/D = locateUID(linked_gun_UID)
+	if(!D)
+		return
+	D.unlink()
+	atom_say("EMP detected. Connection to revolver tracking system lost.")
```

Comment:
It is very very likely that both the pinpointer and det gun will be carried by the same person and therefor will both be EMP'd together. Consequently both the pinpointer and det gun will call `atom_say` so you may want to take that into consideration if it's a bit of a runechat spam along with other EMP effects.

Reply:
Main issue is I want it to be clear to whoever has the items that both are unlinked. While det often holds their pointer, detective has also been giving the pointer to officers, so I want it to be clear to both if the emp effect happens.